### PR TITLE
Update gateway/request to use Promise API from undici

### PR DESCRIPTION
**Pull request by radomird**
Closes: #651

### CODE REVIEW

File: lib/gateway/request.js [hunk 1]
```diff
@@ -37,19 +37,31 @@ function buildRequest (opts) {
...
-      headers: {
-        ...rewriteHeaders(opts.originalRequestHeaders, opts.context),
-        ...opts.headers
-      },
-      body: opts.body
-    }, (err, res) => {
-      if (err) return done(err)
-      done(null, { statusCode: res.statusCode, headers: res.headers, stream: res.body })
-    })
+  async function request (opts, done) {
```
Comment (mcollina) [inline lib/gateway/request.js:40]: remove the done callback completely

**Comment by radomird** [comment]
`done` callback was removed completely. 

@mcollina: Do I need to add something to the documentation, as this functionality was not documented at all (as far as I can see)?

I will add benchmark results soon.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
No need for docs here. I don't expect this to impact benchmarks either.

**Comment by radomird** [comment]
The PR can be merged.

**Pull request merged**